packages/apollo-codegen-scala/src/codeGeneration.ts

```diff
@@ -443,7 +443,7 @@ function dataContainerDeclaration(
     insideCompanion
   }: {
     name: string;
-    properties: Property[];
+    properties: (Property & ({ name: string } | { responseName: string }))[];
```

Comment:
Would typing `properties` correctly here allow you to remove the `any`s below? Or maybe the typing here isn't wrong, but the `any` is just leftover as an oversight? I'll let you find out what works 😄

I _think_ you want

```javascript
(Property & {
  name?: string;
  responseName?: string;
})[];
```

Reply:
The typing here is a bit weird, I know that at least one of `name` and `responseName` will always be defined, but TypeScript isn't handling that too well. Ideally it would understand that in the situation where `name` is `undefined`, then `responseName` will be defined for sure. But instead neither of the properties show up on the object, so I did the `any` typing.

Now that I think of it, it's probably better just to define the type the way you did, so at least there won't be typos with grabbing values from the fields. I'll add a commit to change that in a bit.

Reply:
I see what you mean - I actually see how your original typing is more correct, but I couldn't make TS happy either in a small repl. Let's keep as is since this is fine for all intents and purposes.